### ISSUE TITLE
att_r Parameter not updating in GATConv

### ISSUE THREAD

**Issue by ranahanocka** [rusty1s/pytorch_geometric#1743]
## 🐛 Bug

When running the example GATConv snippet, there is a parameter called `att_r` which is not being updated.

## To Reproduce
Run the following minimal example (taken from the Docs), which just prints which modules have been modified:

```bash
import os.path as osp

import torch
import torch.nn.functional as F
from torch_geometric.datasets import Planetoid
import torch_geometric.transforms as T
from torch_geometric.nn import GATConv
import numpy as np

dataset = 'Cora'
path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', dataset)
dataset = Planetoid(path, dataset, transform=T.NormalizeFeatures())
data = dataset[0]


class Net(torch.nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = GATConv(dataset.num_features, 8, heads=8, dropout=0.6)
        # On the Pubmed dataset, use heads=8 in conv2.
        self.conv2 = GATConv(8 * 8, dataset.num_classes, heads=1, concat=False,
                             dropout=0.6)

    def forward(self):
        x = F.dropout(data.x, p=0.6, training=self.training)
        x = F.elu(self.conv1(x, data.edge_index))
        x = F.dropout(x, p=0.6, training=self.training)
        x = self.conv2(x, data.edge_index)
        return F.log_softmax(x, dim=1)


# device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
device = torch.device('cpu')
model, data = Net().to(device), data.to(device)
optimizer = torch.optim.Adam(model.parameters(), lr=0.005, weight_decay=5e-4)

def get_stats(net):
    a = [param for name, param in net.named_parameters()]
    a_sums = [ai.sum().item() for ai in a]
    a_norms = [ai.norm().item() for ai in a]
    return a, a_sums, a_norms

def train():
    model.train()
    optimizer.zero_grad()
    F.nll_loss(model()[data.train_mask], data.y[data.train_mask]).backward()
    optimizer.step()


def test():
    model.eval()
    logits, accs = model(), []
    for _, mask in data('train_mask', 'val_mask', 'test_mask'):
        pred = logits[mask].max(1)[1]
        acc = pred.eq(data.y[mask]).sum().item() / mask.sum().item()
        accs.append(acc)
    return accs

names = [name for name, param in model.named_parameters()]
before, before_sums, before_norms = get_stats(model)
for epoch in range(1, 200):
    train()
    log = 'Epoch: {:03d}, Train: {:.4f}, Val: {:.4f}, Test: {:.4f}'
    print(log.format(epoch, *test()))

after, after_sums, after_norms = get_stats(model)

diff_norms = [before_norms[i] == after_norms[i] for i in range(len(after_norms))]
diff_sums = [before_sums[i] == after_sums[i] for i in range(len(after_sums))]
mask = np.bitwise_and(diff_norms, diff_sums)
diff = np.where(mask)[0]
print([names[idx] for idx in diff])
print(f'{mask.sum()} out of {len(mask)} did not change')
```

## Expected behavior
I expected att_r to change (note that att_l does), since it is is a nn.Parameter. Maybe I missed something though. Note that the accuracy does get very high, and indeed most parameters are being tuned. I also noticed there is no grad there after .backward() call.

## Environment

* OS: Ubuntu 16.04 & 18.04
* Python version: 3.6.8
* PyTorch version: 1.6
* CUDA/cuDNN version: 10.1 / 10.2

**Comment by rusty1s** [comment]
Thanks for this issue. This is already fixed in master, see [here](https://github.com/rusty1s/pytorch_geometric/commit/923acf947fd0dc170a0ae384289a5b60eca22a45#diff-82ea677a2863fbf09ce4b99505c168e4cd69829d4e17e559d6737eb36d94f234).

**Comment by ranahanocka** [comment]
Great -- I can confirm that after upgrading it is fixed. Required uninstalling first:

```
pip uninstall torch-geometric
pip install git+https://github.com/rusty1s/pytorch_geometric.git
```
